test/foundry-tests/PoolManager.t.sol

```diff
@@ -1023,6 +1039,213 @@ contract PoolManagerTest is Test, Deployers, TokenFixture, GasSnapshot, IERC1155
+        modifyPositionRouter.modifyPosition(key, params, ZERO_BYTES);
+        takeRouter.take(key, 1, 1); // assertions inside takeRouter
+    }
+
+    function test_take_succeedsWithPoolWithLiquidityWithNativeToken() public {
+        PoolKey memory key = PoolKey({
+            currency0: Currency.wrap(address(0)),
+            currency1: currency1,
+            fee: 100,
+            hooks: IHooks(address(0)),
+            tickSpacing: 10
+        });
+        manager.initialize(key, SQRT_RATIO_1_1, ZERO_BYTES);
+
+        IPoolManager.ModifyPositionParams memory params = IPoolManager.ModifyPositionParams(-60, 60, 100);
+        modifyPositionRouter.modifyPosition{value: 100}(key, params, ZERO_BYTES);
+        takeRouter.take{value: 1}(key, 1, 1); // assertions inside takeRouter
+    }
+
+    function test_setProtocolFee_updatesProtocolFeeForInitializedPool() public {
+        uint24 protocolFee = 4;
+
+        PoolKey memory key =
+            PoolKey({currency0: currency0, currency1: currency1, fee: 100, hooks: IHooks(address(0)), tickSpacing: 10});
+        manager.initialize(key, SQRT_RATIO_1_1, ZERO_BYTES);
+
+        (Pool.Slot0 memory slot0,,,) = manager.pools(key.toId());
+        assertEq(slot0.protocolFees, 0);
+        manager.setProtocolFeeController(IProtocolFeeController(address(feeController)));
+        feeController.setSwapFeeForPool(key.toId(), uint16(protocolFee));
+
+        vm.expectEmit(false, false, false, true);
+        emit ProtocolFeeUpdated(key.toId(), protocolFee << 12);
+        manager.setProtocolFees(key);
+    }
+
+    function test_collectProtocolfees_initializesWithProtocolFeeIfCalled() public {
+        uint24 protocolFee = 260; // 0x 0001 00 00 0100
+        PoolKey memory key =
+            PoolKey({currency0: currency0, currency1: currency1, fee: 100, hooks: IHooks(address(0)), tickSpacing: 10});
+        manager.setProtocolFeeController(IProtocolFeeController(address(feeController)));
+        feeController.setSwapFeeForPool(key.toId(), uint16(protocolFee));
+
+        manager.initialize(key, SQRT_RATIO_1_1, ZERO_BYTES);
+        (Pool.Slot0 memory slot0,,,) = manager.pools(key.toId());
+        assertEq(slot0.protocolFees, protocolFee << 12);
+    }
+
+    function test_collectProtocolfees_ERC20_allowsOwnerToAccumulateFees() public {
+        uint24 protocolFee = 260; // 0x 0001 00 00 0100
+        uint256 expectedFees = 7;
+
+        PoolKey memory key =
+            PoolKey({currency0: currency0, currency1: currency1, fee: 3000, hooks: IHooks(address(0)), tickSpacing: 10});
+        manager.setProtocolFeeController(IProtocolFeeController(address(feeController)));
+        feeController.setSwapFeeForPool(key.toId(), uint16(protocolFee));
+
+        manager.initialize(key, SQRT_RATIO_1_1, ZERO_BYTES);
+        (Pool.Slot0 memory slot0,,,) = manager.pools(key.toId());
+        assertEq(slot0.protocolFees, protocolFee << 12);
```

Comment:
would add the assertion that the withdraw protocol fee is 0

Reply:
todo?